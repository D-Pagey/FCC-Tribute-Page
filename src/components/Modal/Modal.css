.modal-btn {
  background-color: transparent;
  border: none;
  margin: .5rem .5rem 0 0; }
  @media screen and (min-width: 430px) {
    .modal-btn {
      margin: 1rem 1rem 0 0; } }
  @media screen and (min-width: 690px) {
    .modal-btn {
      margin: 1.25em 1.5em 0 0; } }
  @media screen and (min-width: 900px) {
    .modal-btn {
      margin: 1.5em 2em 0 0; } }
  @media screen and (min-width: 1400px) {
    .modal-btn {
      margin: 2rem 2rem 0 0; } }
  @media screen and (min-width: 2000px) {
    .modal-btn {
      margin: 2.5em 3em 0 0; } }

.material-icons.md-14 {
  font-size: 14px;
  opacity: .5; }
  @media screen and (min-width: 430px) {
    .material-icons.md-14 {
      font-size: 20px; } }
  @media screen and (min-width: 690px) {
    .material-icons.md-14 {
      font-size: 24px; } }
  @media screen and (min-width: 900px) {
    .material-icons.md-14 {
      font-size: 30px; } }
  @media screen and (min-width: 2000px) {
    .material-icons.md-14 {
      font-size: 40px; } }

.modal-about {
  border: 2px solid black;
  border-radius: 5px;
  margin: 0 auto;
  max-width: 450px;
  padding: .5em;
  top: 15%;
  width: 75%; }
  @media screen and (min-width: 430px) {
    .modal-about {
      width: 70%; } }
  @media screen and (min-width: 690px) {
    .modal-about {
      width: 60%; } }
  @media screen and (min-width: 900px) {
    .modal-about {
      width: 55%; } }
  @media screen and (min-width: 1400px) {
    .modal-about {
      width: 50%; } }

.modal-about::backdrop {
  background: darkgray;
  opacity: .85; }

.modal-cancel {
  background-color: transparent;
  border: none;
  float: right; }

.material-icons.sm-12 {
  font-size: 12px; }
  @media screen and (min-width: 900px) {
    .material-icons.sm-12 {
      font-size: 30px; } }

.modal-title {
  color: royalblue;
  font-size: 1rem;
  padding: .5em;
  text-decoration: underline;
  width: 7em; }
  @media screen and (min-width: 900px) {
    .modal-title {
      font-size: 1.25em; } }

.user-stories {
  font-style: italic;
  padding: .5em 2em .5em 2.5em; }

@media screen and (min-width: 2000px) {
  .story {
    font-size: 1.25em; } }

.story:nth-of-type(2) {
  margin-top: 1em; }

.modal-text {
  margin: .5em; }
  @media screen and (min-width: 2000px) {
    .modal-text {
      font-size: 1.25rem; } }

.link {
  color: royalblue;
  text-decoration: none; }

.ghLink {
  text-decoration: none;
  color: royalblue;
}
