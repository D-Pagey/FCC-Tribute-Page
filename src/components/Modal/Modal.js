import React, { Component } from 'react';

import './Modal.css';

class Modal extends Component {
  constructor(props) {
    super(props);

    this.state = {
      modal: false
    }

    this.handleClick = this.handleClick.bind(this);
  }

  handleClick() {
    if (this.state.modal === false) {
      this.setState({
        modal: true
      });
    } else {
      this.setState({
        modal: false
      })
    }
  }

  render() {

    return (
      <div className="modal-container">
              <button type="button" name="button" className="modal-btn">
                <i className="material-icons md-14" onClick={this.handleClick}>info_outline</i></button>
              <dialog className="modal-about" open={this.state.modal}>
                <button type="button" name="button" className="modal-cancel">
                  <i className="material-icons sm-12" onClick={this.handleClick}>clear</i></button>
                <h4 className="modal-title">Tribute Page</h4>
                <p className="modal-text copy">This is a basic Front End Developer project
                  from the FreeCodeCamp curriculum. The user stories to be completed are:</p>
                <ul className="user-stories">
                  <li className="story copy">I can view a tribute page with an image and text.</li>
                  <li className="story copy">I can click on a link that will take me to an
                    external website with further information on the topic.</li>
                </ul>
                <p className="modal-text copy">This project was coded in React using Create-React-App.
                  The code can be seen on <a href="https://github.com/D-Pagey/Tribute" target="_blank" rel='noreferrer noopener' className='ghLink'>Github</a>.</p>
              </dialog>
            </div>
    )
  }
}

export default Modal;
