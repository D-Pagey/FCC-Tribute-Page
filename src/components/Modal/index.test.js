import React from 'react';
import { shallow } from 'enzyme';

import Modal from '.';

it('renders Modal', () => {
  const wrapper = shallow(
    <Modal />,
  );
  expect(wrapper).toMatchSnapshot();
});
