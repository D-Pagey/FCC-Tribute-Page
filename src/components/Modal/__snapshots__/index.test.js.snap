// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Modal 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Modal />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="modal-btn"
          name="button"
          type="button"
>
          <i
                    className="material-icons md-14"
                    onClick={[Function]}
                    role="button"
                    tabIndex={0}
          >
                    info_outline
          </i>
</button>,
        <dialog
          className="modal-about"
          open={false}
>
          <button
                    className="modal-cancel"
                    name="button"
                    type="button"
          >
                    <i
                              className="material-icons sm-12"
                              onClick={[Function]}
                              role="button"
                              tabIndex={0}
                    >
                              clear
                    </i>
          </button>
          <h4
                    className="modal-title"
          >
                    Tribute Page
          </h4>
          <p
                    className="modal-text"
          >
                    This is a basic Front End Developer project from the FreeCodeCamp curriculum. The user stories to be completed are:
          </p>
          <ul
                    className="user-stories"
          >
                    <li
                              className="story"
                    >
                              I can view a tribute page with an image and text.
                    </li>
                    <li
                              className="story"
                    >
                              I can click on a link that will take me to an external website with further information on the topic.
                    </li>
          </ul>
          <p
                    className="modal-text"
          >
                    This project was coded in React using Create-React-App. The code can be seen on
                    <a
                              className="ghLink"
                              href="https://github.com/D-Pagey/Tribute"
                              rel="noreferrer noopener"
                              target="_blank"
                    >
                               Github
                    </a>
                    .
          </p>
</dialog>,
      ],
      "className": "modal-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <i
            className="material-icons md-14"
            onClick={[Function]}
            role="button"
            tabIndex={0}
>
            info_outline
</i>,
          "className": "modal-btn",
          "name": "button",
          "type": "button",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "info_outline",
            "className": "material-icons md-14",
            "onClick": [Function],
            "role": "button",
            "tabIndex": 0,
          },
          "ref": null,
          "rendered": "info_outline",
          "type": "i",
        },
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              className="modal-cancel"
              name="button"
              type="button"
>
              <i
                            className="material-icons sm-12"
                            onClick={[Function]}
                            role="button"
                            tabIndex={0}
              >
                            clear
              </i>
</button>,
            <h4
              className="modal-title"
>
              Tribute Page
</h4>,
            <p
              className="modal-text"
>
              This is a basic Front End Developer project from the FreeCodeCamp curriculum. The user stories to be completed are:
</p>,
            <ul
              className="user-stories"
>
              <li
                            className="story"
              >
                            I can view a tribute page with an image and text.
              </li>
              <li
                            className="story"
              >
                            I can click on a link that will take me to an external website with further information on the topic.
              </li>
</ul>,
            <p
              className="modal-text"
>
              This project was coded in React using Create-React-App. The code can be seen on
              <a
                            className="ghLink"
                            href="https://github.com/D-Pagey/Tribute"
                            rel="noreferrer noopener"
                            target="_blank"
              >
                             Github
              </a>
              .
</p>,
          ],
          "className": "modal-about",
          "open": false,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <i
                className="material-icons sm-12"
                onClick={[Function]}
                role="button"
                tabIndex={0}
>
                clear
</i>,
              "className": "modal-cancel",
              "name": "button",
              "type": "button",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "clear",
                "className": "material-icons sm-12",
                "onClick": [Function],
                "role": "button",
                "tabIndex": 0,
              },
              "ref": null,
              "rendered": "clear",
              "type": "i",
            },
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Tribute Page",
              "className": "modal-title",
            },
            "ref": null,
            "rendered": "Tribute Page",
            "type": "h4",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "This is a basic Front End Developer project from the FreeCodeCamp curriculum. The user stories to be completed are:",
              "className": "modal-text",
            },
            "ref": null,
            "rendered": "This is a basic Front End Developer project from the FreeCodeCamp curriculum. The user stories to be completed are:",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <li
                  className="story"
>
                  I can view a tribute page with an image and text.
</li>,
                <li
                  className="story"
>
                  I can click on a link that will take me to an external website with further information on the topic.
</li>,
              ],
              "className": "user-stories",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "I can view a tribute page with an image and text.",
                  "className": "story",
                },
                "ref": null,
                "rendered": "I can view a tribute page with an image and text.",
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "I can click on a link that will take me to an external website with further information on the topic.",
                  "className": "story",
                },
                "ref": null,
                "rendered": "I can click on a link that will take me to an external website with further information on the topic.",
                "type": "li",
              },
            ],
            "type": "ul",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "This project was coded in React using Create-React-App. The code can be seen on",
                <a
                  className="ghLink"
                  href="https://github.com/D-Pagey/Tribute"
                  rel="noreferrer noopener"
                  target="_blank"
>
                   Github
</a>,
                ".",
              ],
              "className": "modal-text",
            },
            "ref": null,
            "rendered": Array [
              "This project was coded in React using Create-React-App. The code can be seen on",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": " Github",
                  "className": "ghLink",
                  "href": "https://github.com/D-Pagey/Tribute",
                  "rel": "noreferrer noopener",
                  "target": "_blank",
                },
                "ref": null,
                "rendered": " Github",
                "type": "a",
              },
              ".",
            ],
            "type": "p",
          },
        ],
        "type": "dialog",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="modal-btn"
            name="button"
            type="button"
>
            <i
                        className="material-icons md-14"
                        onClick={[Function]}
                        role="button"
                        tabIndex={0}
            >
                        info_outline
            </i>
</button>,
          <dialog
            className="modal-about"
            open={false}
>
            <button
                        className="modal-cancel"
                        name="button"
                        type="button"
            >
                        <i
                                    className="material-icons sm-12"
                                    onClick={[Function]}
                                    role="button"
                                    tabIndex={0}
                        >
                                    clear
                        </i>
            </button>
            <h4
                        className="modal-title"
            >
                        Tribute Page
            </h4>
            <p
                        className="modal-text"
            >
                        This is a basic Front End Developer project from the FreeCodeCamp curriculum. The user stories to be completed are:
            </p>
            <ul
                        className="user-stories"
            >
                        <li
                                    className="story"
                        >
                                    I can view a tribute page with an image and text.
                        </li>
                        <li
                                    className="story"
                        >
                                    I can click on a link that will take me to an external website with further information on the topic.
                        </li>
            </ul>
            <p
                        className="modal-text"
            >
                        This project was coded in React using Create-React-App. The code can be seen on
                        <a
                                    className="ghLink"
                                    href="https://github.com/D-Pagey/Tribute"
                                    rel="noreferrer noopener"
                                    target="_blank"
                        >
                                     Github
                        </a>
                        .
            </p>
</dialog>,
        ],
        "className": "modal-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <i
              className="material-icons md-14"
              onClick={[Function]}
              role="button"
              tabIndex={0}
>
              info_outline
</i>,
            "className": "modal-btn",
            "name": "button",
            "type": "button",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "info_outline",
              "className": "material-icons md-14",
              "onClick": [Function],
              "role": "button",
              "tabIndex": 0,
            },
            "ref": null,
            "rendered": "info_outline",
            "type": "i",
          },
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                className="modal-cancel"
                name="button"
                type="button"
>
                <i
                                className="material-icons sm-12"
                                onClick={[Function]}
                                role="button"
                                tabIndex={0}
                >
                                clear
                </i>
</button>,
              <h4
                className="modal-title"
>
                Tribute Page
</h4>,
              <p
                className="modal-text"
>
                This is a basic Front End Developer project from the FreeCodeCamp curriculum. The user stories to be completed are:
</p>,
              <ul
                className="user-stories"
>
                <li
                                className="story"
                >
                                I can view a tribute page with an image and text.
                </li>
                <li
                                className="story"
                >
                                I can click on a link that will take me to an external website with further information on the topic.
                </li>
</ul>,
              <p
                className="modal-text"
>
                This project was coded in React using Create-React-App. The code can be seen on
                <a
                                className="ghLink"
                                href="https://github.com/D-Pagey/Tribute"
                                rel="noreferrer noopener"
                                target="_blank"
                >
                                 Github
                </a>
                .
</p>,
            ],
            "className": "modal-about",
            "open": false,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <i
                  className="material-icons sm-12"
                  onClick={[Function]}
                  role="button"
                  tabIndex={0}
>
                  clear
</i>,
                "className": "modal-cancel",
                "name": "button",
                "type": "button",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "clear",
                  "className": "material-icons sm-12",
                  "onClick": [Function],
                  "role": "button",
                  "tabIndex": 0,
                },
                "ref": null,
                "rendered": "clear",
                "type": "i",
              },
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Tribute Page",
                "className": "modal-title",
              },
              "ref": null,
              "rendered": "Tribute Page",
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "This is a basic Front End Developer project from the FreeCodeCamp curriculum. The user stories to be completed are:",
                "className": "modal-text",
              },
              "ref": null,
              "rendered": "This is a basic Front End Developer project from the FreeCodeCamp curriculum. The user stories to be completed are:",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <li
                    className="story"
>
                    I can view a tribute page with an image and text.
</li>,
                  <li
                    className="story"
>
                    I can click on a link that will take me to an external website with further information on the topic.
</li>,
                ],
                "className": "user-stories",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "I can view a tribute page with an image and text.",
                    "className": "story",
                  },
                  "ref": null,
                  "rendered": "I can view a tribute page with an image and text.",
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "I can click on a link that will take me to an external website with further information on the topic.",
                    "className": "story",
                  },
                  "ref": null,
                  "rendered": "I can click on a link that will take me to an external website with further information on the topic.",
                  "type": "li",
                },
              ],
              "type": "ul",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "This project was coded in React using Create-React-App. The code can be seen on",
                  <a
                    className="ghLink"
                    href="https://github.com/D-Pagey/Tribute"
                    rel="noreferrer noopener"
                    target="_blank"
>
                     Github
</a>,
                  ".",
                ],
                "className": "modal-text",
              },
              "ref": null,
              "rendered": Array [
                "This project was coded in React using Create-React-App. The code can be seen on",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": " Github",
                    "className": "ghLink",
                    "href": "https://github.com/D-Pagey/Tribute",
                    "rel": "noreferrer noopener",
                    "target": "_blank",
                  },
                  "ref": null,
                  "rendered": " Github",
                  "type": "a",
                },
                ".",
              ],
              "type": "p",
            },
          ],
          "type": "dialog",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
